pkg/maas/scope/cluster.go

```diff
@@ -118,7 +119,18 @@ func (s *ClusterScope) SetDNSName(dnsName string) {
 // GetDNSName sets the Network systemID in spec.
 // This can't do a lookup on Status.Network.DNSDomain name since it's derviced from here
 func (s *ClusterScope) GetDNSName() string {
-	return fmt.Sprintf("%s.%s", s.Cluster.Name, s.MaasCluster.Spec.DNSDomain)
+	if !s.Cluster.Spec.ControlPlaneEndpoint.IsZero() {
+		return s.Cluster.Spec.ControlPlaneEndpoint.Host
+	}
+
+	if s.MaasCluster.Status.Network.DNSName != "" {
+		return s.MaasCluster.Status.Network.DNSName
+	}
+
+	dnsName := fmt.Sprintf("%s-%s.%s", s.Cluster.Name, uuid.New().String(), s.MaasCluster.Spec.DNSDomain)
```

Comment:
a full uuid might be too long, we could just use a 4-8  char suffix.

Reply:
Ack now we use last 13 charcters -avbcbdbcbbcd along with the hyphen

Reply:
changed to use 6 char suffix.